### PR TITLE
Fix 2746:  Add a 'skip to content' link

### PR DESCRIPTION
**Pull request by bimbolabuari**
# Description of Issue

Screen readers tend to announce all of the links on the page to the user, in order to help with accessibility to bypass parts of each page, for example, navigation links in the mode-bar, filter-bar in the toolbar, and get to main content, skip to content is needed.
Skip to content provides help with the structure of the page bypassing blocks of content, such as navigation menus and page headers, to get to the main content of the page. The other benefit is that sighted users who rely on only the keyboard for navigation can also quickly bypass navigation containing many links.

## Proposed Solution

Skip to content is implemented in the upper left corner that says “Skip to Content.”, It is hidden and then brings into view when it is focused or active. It's the first thing that appears after the page is being loaded and tabbed on, it skips to the first postcard which is the most important of the page on the data view. it is only visible when tabbed on importantly for screen-readers only and hiding the skip-to-content link to everybody else.

## Testing

The testing was done manually by tabbing on the page which in turn activates the skip to content button as shown in the screenshot section below. The `gulp test` command was run to make sure no test is broken/affected by the changes.

## Testing checklist
- [x] I certify that I ran my checklist

### Post views

 - [ ] Go to data posts
 - [ ] Tab on page
 - [ ] "Skip to content" will be visible on top left side
 - [ ] Tab on "skip to content" button
 - [ ] Focus is on "skip to content" link
 - [ ] Press tab or space bar
 - [ ] Focus is on first post card

## Screenshots

|  Current        |    Proposed changes    |
| --------- |:----------:|
|     skip-to-content-pp         |     skip-to-content-cp       |

Fixes ushahidi/platform#2746

Ping @ushahidi/platform @Angamanga @rowasc

### CODE REVIEW

File: app/main/posts/views/card.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <div class="listing-item-select">
         <input ng-show="canSelect" type="checkbox" checklist-value="post.id" checklist-model="selectedPosts" ng-click="stopClickPropagation($event)">
     </div>
-    <div class="postcard-body" ng-click="clickAction($event)">
+    <div class="postcard-body" tabindex="0" role="button" ng-click="clickAction($event)">
```
Comment (bimbolabuari) [inline app/main/posts/views/card.html:6]: @Angamanga  @rowasc, Please can I change this to button tag as it's not clickable(using space-tab or enter) for accessibility, I tried changing it but the styles are not rendering well, I need your permission to override it 

## View with div element
button-cc

## View with button element

button-pc

With just space bar and enter key I was able to tab through the post on button element.

**Comment by bimbolabuari** [comment]
@Angamanga @rowasc, I am having a little pitfall with this proposed skip to content. When data is clicked, the next tab won't get to skip-to-content unless shift-tab is clicked to go back to the skip-to-content link or refreshed, I want to ask if there is a better approach I can take.

**Comment by bimbolabuari** [comment]
@Angamanga @rowasc, Please could you help me review this

File: app/main/posts/views/card.html (hunk 1 above)
Comment (Angamanga) [reply app/main/posts/views/card.html:6]: @bimbolabuari Changing all the cards to buttons is quite a big change and requires some speccing before we go on that route. It is something I am thinking dealing with during the internship. For now, you could check the examples here: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#Examples to figure out how to make the div "clickable" with the space and enter-keys 🙏

File: sass/overrides/_skip-to-content.scss [hunk 2]
```diff
@@ -0,0 +1,18 @@
+.skip-to-content {
```
Comment (Angamanga) [inline sass/overrides/_skip-to-content.scss:1]: Css should live in the pattern-library. The overrides here is only for adjusting css connected to different libraries we are using. The pattern-library lives in this repository: https://github.com/ushahidi/platform-pattern-library/, you can find guidelines on how to set it up here: https://docs.ushahidi.com/platform-developer-documentation/development-and-code/setup_alternatives/setting-up-the-pattern-library-for-development

**Review by Angamanga — CHANGES_REQUESTED** [review]
Nice work! This is a very good improvement! I added a suggestion on how to deal with the div that behaves like a button in the code. I also need you to move the css to the Pattern-library. Once you have set it up and added the css, you create a pull request and add a link to that pr in the description of this pull request. Thank you for working on this 🎉

File: app/main/posts/views/card.html (hunk 1 above)
Comment (bimbolabuari) [reply app/main/posts/views/card.html:6]: @Angamanga, I will do this ASAP, thanks for the feedback!

File: sass/overrides/_skip-to-content.scss (hunk 2 above)
Comment (bimbolabuari) [reply sass/overrides/_skip-to-content.scss:1]: @Angamanga, I will move the styles to the pattern library ASAP
thanks!

File: app/main/posts/views/card.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <div class="listing-item-select">
         <input ng-show="canSelect" type="checkbox" checklist-value="post.id" checklist-model="selectedPosts" ng-click="stopClickPropagation($event)">
     </div>
-    <div class="postcard-body" ng-click="clickAction($event)">
+    <div class="postcard-body" tabindex="0" role="button" ng-keypress="clickAction($event)" ng-click="clickAction($event)">
```
Comment (bimbolabuari) [inline app/main/posts/views/card.html:6]: @Angamanga  @rowasc, I was able to make some changes using angularJS ng-keypress attribute to make the div element clickable

|   Proposed changes   | 
| --------------------------------- |
|  post-onkeypress  |  

Please help me review will appreciate your feedback!

File: app/main/posts/views/post-card.directive.js [hunk 4]
```diff
@@ -47,6 +47,9 @@ function PostCardDirective(FormEndpoint, PostLockService, $rootScope) {
                     // But ignore the action
                     return;
                 }
+                if (evt && evt.key && (evt.key !== ' ' && evt.key !== 'Enter' && evt.key !== 'Spacebar')) {
```
Comment (bimbolabuari) [inline app/main/posts/views/post-card.directive.js:50]: @Angamanga  @rowasc, the ng-keypress attribute was applicable to all keys so I wrote this conditional statement so as to limit its accessibility to only enter or spacebar key

**Comment by bimbolabuari** [comment]
@Angamanga @rowasc, I made a pull request on skip-to-content styles [on platform-pattern-library](https://github.com/ushahidi/platform-pattern-library/pull/260)

**Comment by bimbolabuari** [comment]
Pinging @Angamanga @rowasc

**Review by Angamanga — APPROVED** [review]
Works fine now! Thank you for your contribution!

**Comment by Angamanga** [comment]
@Obadha2 This one is ready for QA but need a new release from the PL. Ping me once you are ready to QA and I'll release it!

**Comment by Obadha2** [comment]
@Angamanga I'm ready. 🙂

**Comment by Angamanga** [comment]
@Obadha2 Here we go: https://github.com/ushahidi/platform-client/pull/1629

**Comment by Angamanga** [comment]
@Obadha2 QA this one first: https://github.com/ushahidi/platform-client/pull/1606 so we go in order with the PL-versions (that one requires version 4.4.4-pre.5, this pr use 4.4.4-pre.6)

**Comment by Obadha2** [comment]
QA'd, passes. 👍

**Pull request merged**